lib/authentication/auth_web.js

```diff
@@ -74,29 +77,47 @@ function AuthWeb(connectionConfig, httpClient, webbrowser) {
     // Use a free random port and set to no backlog
     server.listen(0, 0);
 
-    // Step 1: query Snowflake to obtain SSO url
-    const ssoData = await ssoUrlProvider.getSSOURL(authenticator,
-      serviceName,
-      account,
-      server.address().port,
-      username,
-      host);
-
-    proofKey = ssoData['proofKey'];
-
+    if (connectionConfig.getDisableConsoleLogin()) {
+      // Step 1: query Snowflake to obtain SSO url
+      const ssoData = await ssoUrlProvider.getSSOURL(authenticator,
+        serviceName,
+        account,
+        server.address().port,
+        username,
+        host);
+
+      proofKey = ssoData['proofKey'];
+      loginUrl = ssoData['ssoUrl'];
+    } else {
+      proofKey = this.generateProofKey();
+      loginUrl = this.getLoginUrl(username, proofKey, server.address().port);
+    }
+   
     // Step 2: validate URL
```

Comment:
perhaps instead of these comments `// Step # (...)` we can use functions with descriptive naming?

Reply:
I think your opinion is reasonable. These implementations were done in the past. I think it would be better to change the function name in the different PR.